wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -380,8 +382,17 @@ instance ToSchema (V1 Core.Coin) where
             & type_ .~ SwaggerNumber
             & maximum_ .~ Just (fromIntegral Core.maxCoinVal)
 
+instance ToHttpApiData Core.Coin where
+    toQueryParam = pretty . Core.coinToInteger
+
+instance FromHttpApiData Core.Coin where
+    parseUrlPiece p = do
+        c <- Core.Coin <$> parseQueryParam p
+        Core.checkCoin c
+        pure c
```

Comment:
We gotta need a roundtrip Aeson tests for that one :+1:

Reply:
certainly

Reply:
there is one already in `wallet-new/test/MarshallingSpec.hs` line 95:

Reply:
Sorry, I meant the `httpAPIData` round trip of course :/ ... 

I didn't realize at first this was also (again) an orphan instance. I think we want instances to be on `V1 Core.Coin` here (like the others above). In such case (as for aeson instances, there's no point of doing a rountrip test on the non-wrapped type (`Core.Coin`) itself, because we do want to test and use the wrapped instance. And chances are that there's not even an instance defined for the unwrapped type, which is good. 

May you do this change :pray: ?

Reply:
will be addressed in https://iohk.myjetbrains.com/youtrack/issue/CO-422 for reasons specified therein